File: ladder-client/Classes/NSFoundation+Extension/LDROperation.m
#import "LDROperation.h"
#import "Reachability.h"


#pragma mark - LDROperation
@implementation LDROperation


#pragma mark - synthesize


#pragma mark - api
- (void)start
{
    if ([[Reachability reachabilityForInternetConnection] currentReachabilityStatus] == NotReachable) {
        [self willChangeValueForKey:@"isFinished"];
        _finished = YES;
        [self didChangeValueForKey:@"isFinished"];
        self.handler(nil,
                     @{},
                     [[NSError alloc] initWithDomain: NSMachErrorDomain
                                                code: 0
                                                userInfo:@{}]);
        return;
    }

    [super start];
}


#pragma mark - private api


@end

File: ladder-client/Classes/NSFoundation+Extension/LDROperation.swift
/*
import Foundation
import ISHTTPOperation

class LDROperation: ISHTTPOperation {

  enum State: String {
    case Ready, Executing, Finished

    var keyPath: String {
      "is" + rawValue
    }
  }

  var state = State.Ready {
    willSet {
      willChangeValue(forKey: newValue.keyPath)
      willChangeValue(forKey: state.keyPath)
    }
    didSet {
      didChangeValue(forKey: oldValue.keyPath)
      didChangeValue(forKey: state.keyPath)
    }
  }
}

extension LDROperation {

  override var isReady: Bool {
    super.isReady && state == .Ready
  }

  override var isExecuting: Bool {
    state == .Executing
  }

  override var isFinished: Bool {
    state == .Finished
  }

  override func start() {
    if let connection = Reachability.forInternetConnection() {
        if connection.currentReachabilityStatus() == NotReachable {
            state = .Finished
            return
        }
    }
    if isCancelled {
      state = .Finished
      return
    }
    main()
    state = .Executing
  }

  override func cancel() {
    state = .Finished
  }
}
*/
